lib/cache/cache.go

```diff
@@ -556,6 +559,19 @@ func (c *Cache) read() (readGuard, error) {
 	}, nil
 }
 
+// kindSubKind is used as a key in maps allowing lookups by a combination of kind and subKind
+type kindSubKind struct {
```

Comment:
This is almost the same as `resourceKind`, but doesn't have `version`. I could use it instead of this type assuming `version` would always be empty, but it didn't seem right.

Maybe if we clean up [that version condition for KindAppServer](https://github.com/gravitational/teleport/blob/2f09551c7cbb95e40b3664ef89718fa6fa4597c7/lib/services/local/events.go#L108) and remove `version` from `resourceKind`, we won't need this extra type.

Reply:
I wouldn't be opposed to that, but we need to decide what to do if in the future we end up wanting to support watching only certain versions or up to certain versions of a resource for the sake of backwards compatibility (if that's how we ended up watching for `KindAppServer` but only version V2).

Reply:
It seems hard to plan for something like that without a specific use case in mind. So my idea was to do a "reset" by erasing all version-specific logic and then re-adding it later when needed.

Reply:
You've convinced me 👍